File: roombase/src/main/java/org/phoneservice/roombase/java/Person.java
package org.phoneservice.roombase.java;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * Created by Administrator on 2020/7/27 0027.
 * 类描述
 * 版本
 */
@Entity //代表此类会映射到SQLite数据库中
public class Person {
    @PrimaryKey(autoGenerate = true)
    private int id;
    @ColumnInfo(name = "person_name")
    private String name;
    @ColumnInfo(name = "person_age")
    private int age;

    Person() {
    }

    Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    void setAge(int age) {
        this.age = age;
    }

    public int getId() {
        return id;
    }

    void setId(int id) {
        this.id = id;
    }
}

File: lifecycles/src/test/java/org/phoneservice/lifecycles/Person.kt
package org.phoneservice.lifecycles

/**
 * Created by Administrator on 2020/7/26 0026.
 * 类描述  构造函数专题：
 * 版本
 */


/**
 * 主构函数=====================================================开始
 */
/**
 * 写法1：
 */
//class Person(name: String, age: Int) {
//}
/**
 * 写法2：target
 */
//class Person(name: String, age: Int) {
//    var name: String = ""
//    var age: Int = 0
//    init {
//        this.name = name
//        this.age = age
//    }
//}
/**
 * 写法3：target
 */
//class Person(var name: String, var age: Int) {
//}
/**
 * 主构函数=====================================================结束
 */


/**
 * 次构函数=====================================================开始
 */
class Person(var name: String, var age: Int) {
    var number: String = ""

    constructor(name: String, age: Int, number: String) : this(name, age) {
        this.number = number
    }
}

